[cyaoxuan] iP

cyaoxuan opened the pull request:
# EepyBot 

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

EepyBot frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use
- eepy 

All you need to do is,
1. download it from [here](https://github.com/cyaoxuan/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

cyaoxuan committed:
Level-0 implemented. Renamed to Eepy

cyaoxuan committed:
Level-0 implemented. Renamed to Eepy

cyaoxuan committed:
Merge branch 'master' of https://github.com/cyaoxuan/ip

cyaoxuan committed:
Level-1 implemented

cyaoxuan committed:
Level-2 imeplemented

cyaoxuan committed:
Level-3 implemented

cyaoxuan committed:
Level-4 implemented

cyaoxuan committed:
A-TextUiTesting done. Indentation fixes

cyaoxuan committed:
Level-5 implemented. Most errors should be handled

cyaoxuan committed:
Level-6 implemented

cyaoxuan committed:
Implement save tasks in hard disk

cyaoxuan committed:
Updata .gitignore

cyaoxuan committed:
Add prompt

cyaoxuan committed:
Merge branch 'branch-Level-7'

cyaoxuan committed:
Include support for Date for Deadline objects

cyaoxuan committed:
Merge branch 'branch-Level-8'

cyaoxuan committed:
Add classes to support OOP principles

cyaoxuan committed:
Put all classes under same duke package

cyaoxuan committed:
Merge branch 'add-gradle-support' of https://github.com/cyaoxuan/ip

cyaoxuan committed:
Add gradle support

cyaoxuan committed:
Test Parser and Task classes

cyaoxuan committed:
Create a JAR file

cyaoxuan committed:
Add JavaDocs

cyaoxuan committed:
Edit code to follow coding standards

cyaoxuan committed:
Add JavaDocs

cyaoxuan committed:
Implement find functionality

cyaoxuan committed:
Merge branch 'branch-A-JavaDoc'

cyaoxuan committed:
Fix merge conflicts due to naming discrepancies

cyaoxuan committed:
Fix merge conflicts

cyaoxuan committed:
Fix merge conflicts

ChuanXinNg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,176 @@
+     * @return process code, where 0 means the program should end and 1 otherwise
+     * @throws DukeException if there are malformed inputs
+     */
+    public static int parseInput(String input, TaskList taskList) throws DukeException {
```
Can checkout the java.util.regex package that is used for working with regular expressions in Java such as java.util.regex.Pattern to make your code more readable and manageable.

ChuanXinNg commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+
+    @Override
+    public String getMessage() {
+        return "⚠ Eek! " + super.getMessage();
```
Creative use of ⚠, good job!

ChuanXinNg commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Chatbot that manages a task list, which is saved into the hard disk.
+ */
+public class Duke {
```
Can change your Duke class to your bot's name, Eepy.

ChuanXinNg reviewed (approved):
I found your code easy to read and understand. Good job!

cyaoxuan committed:
Add GUI

cyaoxuan committed:
Merge branch 'branch-Level-10'

tjingsheng commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,53 @@
+        this.getChildren().setAll(tmp);
+    }
+
+    public static DialogBox getUserDialog(Label l, ImageView iv) {
```
I would prefer to put these function at the top. right after the dialog box constructor

tjingsheng commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,180 @@
+import java.io.File;
+import java.util.Scanner;
+
+import javafx.application.Application;
```
```
I would prefer to arrange the imports as such

package duke;

import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.control.ScrollPane;
import javafx.scene.control.TextField;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;
import javafx.scene.layout.AnchorPane;
import javafx.scene.layout.Region;
import javafx.scene.layout.VBox;
import javafx.stage.Stage;

import java.io.File;
import java.util.Scanner;

```

tjingsheng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,173 @@
+public class Parser {
+    /**
+     * Parses the input gotten from the main menu.
+     * @param input user input
```
```
I would prefer to put the javaDoc as such

    /**
     * Parses the input gotten from the main menu.
     *
     * @param input    user input
     * @param taskList task list managed by the chatbot
     * @return process code, where 0 means the program should end and 1 otherwise
     * @throws DukeException if there are malformed inputs
     */
```

tjingsheng commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,79 @@
+    public static final String FILE_PATH = "./data/duke.txt";
+
+    /**
+     * Loads the existing task list file and creates one if it does not exist.
```
I would prefer to add a line here
```
    /**
     * Loads the existing task list file and creates one if it does not exist.
     *
     * @param taskList task list managed by the chatbot to populate
     * @return process code, where 0 means no file was read / created and 1 otherwise
     */
```

tjingsheng commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,54 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
```
Are these imports used?

tjingsheng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+        return count;
+    }
+
+    /**
```
I would prefer the javaDoc like this
```
    /**
     * Adds a given task to the list.
     *
     * @param t           task to be added
     * @param shouldPrint whether text should be printed, disabled when it is loaded from a file
     * @return string output after adding task
     */
   ```

tjingsheng commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+        System.out.println("Exit program - bye");
+    }
+
+    /**
```
I would prefer the javaDoc like this
```
    /**
     * Initialises the UI, displaying options to the user.
     *
     * @return scanner to get input from user
     */
    public static Scanner initialiseUi() {
        System.out.println("");
        System.out.println("What can I do for you?");
        displayOptions();
        Scanner in = new Scanner(System.in);
        return in;
    }

    /**
     * Takes in user input.
     *
     * @param in scanner to be used to get input
     * @return user input as a string
     */
 ```

cyaoxuan replied:
Yes you feedbacked the right person :D There are 2 rounds of reviews, hence the review above too. Thank you for the feedback!

tjingsheng reviewed (commented):
Your code is very good. I added some nitpicks for the javaDocs and imports order. Keep up the good work. PS, do let me know if i reviewed the correct PR. I see someone has commented on this PR as well :(

cyaoxuan committed:
Modify build.gradle for generation of JAR file

cyaoxuan committed:
Add assertions to ensure key assumptions hold

Have to ensure that integrity of data remains even after methods that
modify the tasks and task lists are called. Failure to ensure this will
lead to the whole program malfunctioning.

Let's
- ensure that parameters passed to other functions are in the correct
format

cyaoxuan committed:
Improve code quality

Refactored code to make it more readable and understandable.

Let's
- ensure SLAP is adhered to by abstracting out file operations as
separate functions dealing with different cases
- Include meaningful comments to make verbose code more understandable

cyaoxuan committed:
Merge pull request #1 from cyaoxuan/branch-A-Assertions

Add assertions to ensure key assumptions hold

cyaoxuan committed:
Merge pull request #2 from cyaoxuan/branch-A-CodeQuality

Improve code quality

cyaoxuan committed:
Add tagging feature

cyaoxuan committed:
Merge branch 'branch-C-Tagging'

# Conflicts:
#	src/main/java/duke/Parser.java

cyaoxuan committed:
Improve code quality

cyaoxuan committed:
Update tests and add FXML

cyaoxuan committed:
Update user guide
